File: gesture/src/main/java/com/omarea/gesture/shell/KeepShellPublic.java
package com.omarea.gesture.shell;

import java.util.List;

/**
 * Created by Hello on 2018/01/23.
 */
public class KeepShellPublic {
    private static KeepShell keepShell = null;

    public static Boolean doCmdSync(List<String> commands) {
        StringBuilder stringBuilder = new StringBuilder();

        for (String cmd : commands) {
            stringBuilder.append(cmd);
            stringBuilder.append("\n\n");
        }

        return doCmdSync(stringBuilder.toString()) != "error";
    }

    //执行脚本
    public static String doCmdSync(String cmd) {
        if (keepShell == null) {
            keepShell = new KeepShell();
        }
        return keepShell.doCmdSync(cmd);
    }

    //执行脚本
    public static Boolean checkRoot() {
        if (keepShell == null) {
            keepShell = new KeepShell();
        }
        return keepShell.checkRoot();
    }

    public static void tryExit() {
        if (keepShell != null) {
            keepShell.tryExit();
            keepShell = null;
        }
    }
}

File: app/src/main/java/com/omarea/shell/KeepShellPublic.kt
package com.omarea.shell

/**
 * Created by Hello on 2018/01/23.
 */
object KeepShellPublic {
    private var keepShell: KeepShell? = null

    //执行脚本
    internal fun doCmdSync(cmd: String): String {
        if (keepShell == null) {
            keepShell = KeepShell()
        }
        return keepShell!!.doCmdSync(cmd)
    }
}
